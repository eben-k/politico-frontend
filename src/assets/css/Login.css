.login-page {
  background: #f5f9f7;
  width: 100%;
  padding: 55px 0px;
  justify-content: center;
  height: 100vh;
  margin-left: auto;
  margin-right: auto;
}
.login-form{
  margin: 30px auto 0 auto;
  width: 400px;
  box-shadow: 1px 2px 5px 3px #9b8b85;
  padding: 10px 0 30px 0;
  background: white;
}
.login-heading {
  color: #7ecddd;
  text-align: center;
}
.login {
  width: 100%;
  display: grid;
  justify-content: center;
}
input[type="text"], input[type="password"], input[type="tel"], input[type="number"]{
  width: 341px;
  height: 50px;
  margin-bottom: 26px;
  border: none;
  background: #f3efef;
  padding-left: 15px;
  outline: none;
}
input[type="submit"] {
  width: 350px;
  height: 50px;
  border: none;
  color: white;
  background: #7ecddd;
  cursor: pointer;
  font-size: 20px;
}
