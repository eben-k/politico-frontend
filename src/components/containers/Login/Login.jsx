import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { loginUser } from '../../../store/modules/auth';

import Layout from '../../presentationals/Layout';

import '../../../assets/css/Login.css';

export const Login = ({ loginUser: handleLogin }) => {
  const [userDetails, setUserDetails] = useState({
    email: '',
    password: '',
  });

  function updateLocalState(event) {
    setUserDetails({ ...userDetails, [event.target.name]: event.target.value });
  }
  async function onSubmit(event) {
    event.preventDefault();
    const loginResponse = await handleLogin(userDetails);
    return loginResponse;
  }
  return (
    <Layout>
      <div className="login-form">
        <h1 className="login-heading">Log In</h1>
        <div className="form">
          <form onSubmit={onSubmit} className="login" action="">
            <div>
              <input
                type="text"
                name="email"
                placeholder="Email"
                required
                onChange={updateLocalState}
              />
            </div>
            <div>
              <input
                type="password"
                name="password"
                placeholder="Password"
                required
                onChange={updateLocalState}
              />
            </div>

            <div>
              <input type="submit" value="LOG IN" />
            </div>

            <div className="forgotpassword">
              <p>
                <a href="reset.html">Forgot Password?</a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </Layout>
  );
};

Login.propTypes = {
  loginUser: PropTypes.func,
};

const mapStateToProps = state => ({
  isLoading: state.auth.isLoading,
  errors: state.auth.errors,
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(
  mapStateToProps,
  { loginUser },
)(Login);
