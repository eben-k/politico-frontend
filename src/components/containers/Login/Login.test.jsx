import React from 'react';
import { shallow } from 'enzyme';

import { Login } from './Login';

const findByTestAttribute = (component, attribute) => {
  const wrapper = component.find(`[data-test='${attribute}']`);
  return wrapper;
};

const component = shallow(<Login />);

describe('Custom Search', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });

  xit('renders the form section', () => {
    expect(findByTestAttribute(component, 'login-form').length).toBe(1);
  });

  xit('renders a submit button', () => {
    expect(findByTestAttribute(component, 'submit-button').length).toBe(1);
  });
});
