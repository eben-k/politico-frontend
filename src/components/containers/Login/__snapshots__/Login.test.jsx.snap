// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Search renders without errors 1`] = `
<Layout>
  <div
    className="login-form"
  >
    <h1
      className="login-heading"
    >
      Log In
    </h1>
    <div
      className="form"
    >
      <form
        action=""
        className="login"
        onSubmit={[Function]}
      >
        <div>
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="text"
          />
        </div>
        <div>
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
        </div>
        <div>
          <input
            type="submit"
            value="LOG IN"
          />
        </div>
        <div
          className="forgotpassword"
        >
          <p>
            <a
              href="reset.html"
            >
              Forgot Password?
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</Layout>
`;
